File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> dp;
        int n = nums.size() - 1;
        for(int i = 0;i<=n; i++){
            if(dp[nums[i]] == 1){
                return true;
            }
            dp[nums[i]] = 1;
        }
        return false;
    }
};

File: Data Structures & Algorithms/remove-element/submission-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int l = 0;
        int r = 0;
        while(l<nums.length) {
            if(nums[l] != val) {
                nums[r] = nums[l];
                r++;
            }
            l++;
        }
        return r;
    }
}